src/compose/images.ts

```diff
@@ -214,8 +213,7 @@ export async function triggerFetch(
 		// Create image on the database if it already exists on the engine
 		await markAsSupervised({ ...image, dockerImageId: img.Id });
 
-		onFinish(true);
-		return;
+		success = true;
```

Comment:
This means it'll now fall through to the normal finish procedure which handles both the `finish` event and the `onFinish` callback

Reply:
This sounds like a good solution, but I think we should probably modify the report function a bit. If the execution gets to the end of the first try, the report will get a 'finish' event without ever triggering a start, which right now will fail. I'm trying to think of a scenario that will get to that run path so I can test.

Reply:
I guess it depends on whether or not we want a report to be triggered when the fetch is skipped due to the image being on disk. I can see value in that too and if that's the case, the change I made to the reportEvent function prevents other issues.

Reply:
I remembered the issue we saw now. The way the `reportEvent` function was implemented, if an event were to be called out of order (e.g. finish without a start), it would create a `runningTask[<imageName>] = undefined` which would cause the `getDownloadingImageNames` function (line 352) to fail.